tracer-core/src/main/java/org/zalando/tracer/Tracer.java

```diff
@@ -31,32 +31,98 @@
 import static com.google.common.collect.Maps.toMap;
 import static java.util.Arrays.asList;
 
+/**
+ * A tracer is a lifecycle manager of one or more {@link Trace traces}. Traces are identified by name, e.g.
+ * {@code X-Trace-ID} and, during an active lifecycle, they can have a value, e.g.
+ * {@code 5dc90f18-231f-11e6-a17d-0f84f5d57f60}. Values can either be provided on a per-lifecycle basis or generated
+ * on-demand by a {@link Generator generator}. Lifecycle events, i.e. start and stop, can be observed by registering
+ * one or more {@link TraceListener listeners}.
+ *
+ * @see Trace
+ * @see Generator
+ * @see TraceListener
+ */
 public interface Tracer {
 
     /**
-     * @throws IllegalStateException
+     * Starts a new trace lifecycle without transporting any trace identifiers.
+     * <p>
+     * The most common use case would be to manage a trace lifecycle in a scheduled background job or during tests.
+     * </p>
+     * <pre>
+     * tracer.start();
+     *
+     * try {
+     *     // do some background work here
+     * } finally {
+     *     tracer.stop();
+     * }
+     * </pre>
+     *
+     * @throws IllegalStateException if this tracer is already started
      */
     default void start() {
-        start(trace -> null);
+        start(name -> null);
     }
 
     /**
-     * @throws IllegalStateException
+     * Starts a new trace lifecycle with the possibility to transport given trace identifiers.
+     * <p>
+     * The most common use case would be to read headers from an incoming HTTP request and hand them over to this
+     * tracer in order to persist them:
+     * </p>
+     * <pre>
+     * tracer.start(request::getHeader)
+     *
+     * try {
+     *     // continue with processing the request
+     * } finally {
+     *     tracer.stop();
+     * }
+     * </pre>
+     *
+     * @param provider a provider for trace identifiers, will be called with individual trace names
```

Comment:
I guess this could have a mention that when the function returns null, a generated value will be used for that trace.